src/centroiders.cpp

```diff
@@ -117,14 +197,33 @@ struct CentroidParams {
     int yMin;
     int yMax;
     int cutoff;
+    std::vector<int> localCutoff;
     bool isValid;
     std::unordered_set<int> checkedIndices;
 };
 
+// int iToDivision
+
+// int Row(i, imageWidth, subdivisions, imageHeight / subdivisions, imageHeight % subdivisions)
+// int Column(i, imageHeight, subdivisions, imageWidth / subdivisions, imageWidth % subdivisions)
+long RowOrColumn(long i, int size, int subdivisions, int div, int leftover) {
+    if(i < (div + 1) * leftover) {
+        return i / (div + 1);
+    } else {
+        return leftover + (i - (div + 1) * leftover) / (div);
+    }
+}
+
+// For a given i and picture dimensions, determines which subdivision i is in (Zero Based)
+long FindSubdivision(long i, int imageWidth, int imageHeight, int subdivisions) {
+    return RowOrColumn(i / imageWidth, imageWidth, subdivisions, imageHeight / subdivisions, imageHeight % subdivisions) * subdivisions + 
+    RowOrColumn(i % imageWidth, imageHeight, subdivisions, imageWidth / subdivisions, imageWidth % subdivisions);
```

Comment:
remove line after `return`.

Reply:
Mark, I believe lines 219 and 220 are technically on the same statement, I just extended the line for readability. I have actually adjusted this and kept a version of it in the edit, so if you still don't want that, I will revert that.

Reply:
I see. It would be more clear if the second line was indented to indicate that it belongs to the previous line. But the most clear solution will be to assign both parts to separate local variables with descriptive names, then return their sum.